### PR TITLE
fix(communities): Naming validation for communities

### PR DESCRIPTION
**Pull request by saledjenic**
I sorted out some validation related stuff. There were some unused and some redundant methods. I guess that validation messages for each input will be uniform now, across these three parts (channel, channel category and community). If it's needed we may easily change text of the error messages (Utils.validateAndReturnError) which will have same pattern now.

Fixes: #2297

### CODE REVIEW

File: ui/imports/Utils.qml [hunk 1]
```diff
@@ -552,6 +552,57 @@ QtObject {
...
     }
 
+    function hasUpperCaseLetter(str) {
+        return (/[A-Z]/.test(str))
+    }
+
+    function convertSpacesToDashesAndUpperToLowerCase(str)
+    {
+        if (str.includes(" "))
+            str = str.replace(" ", "-")
```
Comment (jrainville) [inline ui/imports/Utils.qml:562]: Replace only replaces the first one without using a regex
```suggestion
            str = str.replace(/ /g, "-")
```

**Review by jrainville — CHANGES_REQUESTED** [review]
Very clean!

Requesting changes because the `replace` needs to be fixed, but it's  a minor issue

**Comment by saledjenic** [comment]
@jrainville I fixed the issue you added, you can proceed with review.

**Review by jrainville — APPROVED** [review]

**Comment by anastasiyaig** [comment]
@saledjenic could please check why windows build is not being created here. it is something related to the changes u made?
we can merge after the windows artefact is created

**Comment by jakubgs** [comment]
Okay, i see the error now:
```
-- Selecting Windows SDK version 10.0.10240.0 to target Windows 10.0.19041.
CMake Error at CMakeLists.txt:3 (project):
  Failed to run MSBuild command:

    C:/BuildTools/MSBuild/15.0/Bin/MSBuild.exe

  to get the value of VCTargetsPath:

    Microsoft (R) Build Engine version 15.8.169+g1ccb72aefa for .NET Framework
    Copyright (C) Microsoft Corporation. All rights reserved.

    Build started 7/1/2021 3:38:29 PM.
    Project "C:\Users\jenkins\workspace\desktop_branches_windows_PR-2782\vendor\DOtherSide\build\CMakeFiles\3.20.5\VCTargetsPath.vcxproj" on node 1 (default targets).
    C:\BuildTools\Common7\IDE\VC\VCTargets\Microsoft.Cpp.WindowsSDK.targets(46,5): error MSB8036: The Windows SDK version 10.0.10240.0 was not found. Install the required version of Windows SDK or change the SDK ve
rsion in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\Users\jenkins\workspace\desktop_branches_windows_PR-2782\vendor\DOtherSide\build\CMakeFiles\3.20.5\VCTarg
etsPath.vcxproj]
    Done Building Project "C:\Users\jenkins\workspace\desktop_branches_windows_PR-2782\vendor\DOtherSide\build\CMakeFiles\3.20.5\VCTargetsPath.vcxproj" (default targets) -- FAILED.

    Build FAILED.

    "C:\Users\jenkins\workspace\desktop_branches_windows_PR-2782\vendor\DOtherSide\build\CMakeFiles\3.20.5\VCTargetsPath.vcxproj" (default target) (1) ->
    (_CheckWindowsSDKInstalled target) ->
      C:\BuildTools\Common7\IDE\VC\VCTargets\Microsoft.Cpp.WindowsSDK.targets(46,5): error MSB8036: The Windows SDK version 10.0.10240.0 was not found. Install the required version of Windows SDK or change the SDK 
version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\Users\jenkins\workspace\desktop_branches_windows_PR-2782\vendor\DOtherSide\build\CMakeFiles\3.20.5\VCTa
rgetsPath.vcxproj]

        0 Warning(s)
        1 Error(s)

    Time Elapsed 00:00:00.61


  Exit code: 1
```
Looks like the issue is with the Windows SDK installation.

**Comment by jakubgs** [comment]
We've had this issue before: https://github.com/status-im/status-desktop/pull/2469#issuecomment-830008544

**Comment by jakubgs** [comment]
The error states that:
```
error MSB8036: The Windows SDK version 10.0.10240.0 was not found.
Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution".
```
We CLEARLY install `10.0.10240.0` because that's the version that's selected in our build setup script:
https://github.com/status-im/status-desktop/blob/aa4bdaf9108340dbbf7116cd4a9b0c9aa2ea6a5e/scripts/windows_build_setup.ps1#L45-L47

So the error makes zero sense to me.

**Comment by jakubgs** [comment]
I had some similar issue when I was setting this up in the first place: https://github.com/status-im/status-desktop/issues/595#issuecomment-698421422

Then I just downgrade the Visual C++ Build Tools verison. But we've been using this version for a while and it worked.

I did replace the hosts because of https://github.com/status-im/nim-waku/pull/652#issuecomment-871205562, so I had to reinstall Build Tools, but nothing should have changed about how it is installed.

**Comment by jakubgs** [comment]
There's something seriously wrong with the Build Tools installation process:
```
PS C:\Users\jenkins\Downloads> . .\windows_build_setup.ps1
PS C:\Users\jenkins\Downloads> Install-VC-BuildTools
Downloading Microsoft Visual C++ Build Tools...
Installing Microsoft Visual C++ Build Tools...

Handles  NPM(K)    PM(K)      WS(K)     CPU(s)     Id  SI ProcessName
-------  ------    -----      -----     ------     --  -- -----------
     54       6      892       4096       0.06   6512   0 vs_BuildTools


PS C:\Users\jenkins\Downloads>
admin@windows-02 MINGW64 /c/Users/jenkins/Downloads
$ ls -l /c/BuildTools/
total 0
```

**Comment by jakubgs** [comment]
I don't get it, for some reason the Build Tools executable doesn't even return help:
```
admin@WINDOWS-02 C:\Users\admin\Downloads>.\vs_BuildTools.exe --help

admin@WINDOWS-02 C:\Users\admin\Downloads>
```

**Comment by jakubgs** [comment]
It looks like running the installer leaves a bunch of defunct processes laying around:

image

**Comment by jakubgs** [comment]
I upgraded our Windows Server image to be Windows Server 2019 Core in https://github.com/status-im/infra-ci/commit/3c04e159 and it works now.

**Pull request merged**